2023.5.2

balloob opened the pull request:
- Fix transmission error handling ([@DDanii] - [#91548]) ([transmission docs])
- Fix power sensor state_class in Netatmo integration ([@hex7c0] - [#92468]) ([netatmo docs])
- Fix for SIA Code not being handled well ([@eavanvalkenburg] - [#92469]) ([sia docs])
- Fix scene service examples ([@karwosts] - [#92501]) ([scene docs])
- Fix blocking I/O in the event loop when starting ONVIF ([@bdraco] - [#92518]) ([onvif docs])
- Improve cloud migration ([@emontnemery] - [#92520]) ([cloud docs])
- Bump aioesphomeapi to 13.7.3 to fix disconnecting while handshake is in progress ([@bdraco] - [#92537]) ([esphome docs])
- Handle invalid ZHA cluster handlers ([@puddly] - [#92543]) ([zha docs])
- Allow duplicate state updates when force_update is set on an esphome sensor ([@bdraco] - [#92553]) ([esphome docs])
- Bump elkm1-lib to 2.2.2 ([@bdraco] - [#92560]) ([elkm1 docs])
- Re-run expose entities migration if first time failed ([@balloob] - [#92564]) ([cloud docs])
- Fix hassio get_os_info retry ([@epenet] - [#92569]) ([hassio docs])
- Bump hatasmota to 0.6.5 ([@emontnemery] - [#92585]) ([tasmota docs])
- Improve reliability of ONVIF subscription renewals ([@bdraco] - [#92551]) ([onvif docs])
- Lower scan interval for OpenSky ([@joostlek] - [#92593]) ([opensky docs])
- Update frontend to 20230503.3 ([@bramkragten] - [#92617]) ([frontend docs])
- Fix missing ONVIF events when switching from PullPoint to webhooks ([@bdraco] - [#92627]) ([onvif docs])
- Add support for visitor detections to onvif ([@bdraco] - [#92350]) ([onvif docs])
- Fix duplicate ONVIF sensors ([@bdraco] - [#92629]) ([onvif docs])

[#91548]: https://github.com/home-assistant/core/pull/91548
[#92350]: https://github.com/home-assistant/core/pull/92350
[#92422]: https://github.com/home-assistant/core/pull/92422
[#92468]: https://github.com/home-assistant/core/pull/92468
[#92469]: https://github.com/home-assistant/core/pull/92469
[#92501]: https://github.com/home-assistant/core/pull/92501
[#92513]: https://github.com/home-assistant/core/pull/92513
[#92518]: https://github.com/home-assistant/core/pull/92518
[#92520]: https://github.com/home-assistant/core/pull/92520
[#92537]: https://github.com/home-assistant/core/pull/92537
[#92543]: https://github.com/home-assistant/core/pull/92543
[#92551]: https://github.com/home-assistant/core/pull/92551
[#92553]: https://github.com/home-assistant/core/pull/92553
[#92560]: https://github.com/home-assistant/core/pull/92560
[#92564]: https://github.com/home-assistant/core/pull/92564
[#92569]: https://github.com/home-assistant/core/pull/92569
[#92585]: https://github.com/home-assistant/core/pull/92585
[#92593]: https://github.com/home-assistant/core/pull/92593
[#92617]: https://github.com/home-assistant/core/pull/92617
[#92627]: https://github.com/home-assistant/core/pull/92627
[#92629]: https://github.com/home-assistant/core/pull/92629
[@DDanii]: https://github.com/DDanii
[@balloob]: https://github.com/balloob
[@bdraco]: https://github.com/bdraco
[@bramkragten]: https://github.com/bramkragten
[@eavanvalkenburg]: https://github.com/eavanvalkenburg
[@emontnemery]: https://github.com/emontnemery
[@epenet]: https://github.com/epenet
[@frenck]: https://github.com/frenck
[@hex7c0]: https://github.com/hex7c0
[@joostlek]: https://github.com/joostlek
[@karwosts]: https://github.com/karwosts
[@puddly]: https://github.com/puddly
[accuweather docs]: https://www.home-assistant.io/integrations/accuweather/
[advantage_air docs]: https://www.home-assistant.io/integrations/advantage_air/
[bluetooth docs]: https://www.home-assistant.io/integrations/bluetooth/
[cloud docs]: https://www.home-assistant.io/integrations/cloud/
[elkm1 docs]: https://www.home-assistant.io/integrations/elkm1/
[esphome docs]: https://www.home-assistant.io/integrations/esphome/
[frontend docs]: https://www.home-assistant.io/integrations/frontend/
[hassio docs]: https://www.home-assistant.io/integrations/hassio/
[lifx docs]: https://www.home-assistant.io/integrations/lifx/
[netatmo docs]: https://www.home-assistant.io/integrations/netatmo/
[notion docs]: https://www.home-assistant.io/integrations/notion/
[onvif docs]: https://www.home-assistant.io/integrations/onvif/
[opensky docs]: https://www.home-assistant.io/integrations/opensky/
[scene docs]: https://www.home-assistant.io/integrations/scene/
[sia docs]: https://www.home-assistant.io/integrations/sia/
[tasmota docs]: https://www.home-assistant.io/integrations/tasmota/
[transmission docs]: https://www.home-assistant.io/integrations/transmission/
[zha docs]: https://www.home-assistant.io/integrations/zha/

bdraco committed:
Add support for visitor detections to onvif (#92350)

karwosts committed:
Fix scene service examples (#92501)

hex7c0 committed:
Fix power sensor state_class in Netatmo integration (#92468)

bdraco committed:
Fix blocking I/O in the event loop when starting ONVIF (#92518)

puddly committed:
Handle invalid ZHA cluster handlers (#92543)

* Do not crash on startup when an invalid cluster handler is encountered

* Add a unit test

bdraco committed:
Bump aioesphomeapi to 13.7.3 to fix disconnecting while handshake is in progress (#92537)

Bump aioesphomeapi to 13.7.3

fixes #92432

bdraco committed:
Bump elkm1-lib to 2.2.2 (#92560)

changelog: https://github.com/gwww/elkm1/compare/2.2.1...2.2.2

fixes #92467

bdraco committed:
Allow duplicate state updates when force_update is set on an esphome sensor (#92553)

* Allow duplicate states when force_update is set on an esphome sensor

fixes #91221

* Update homeassistant/components/esphome/entry_data.py

Co-authored-by: pdw-mb <pdw@mythic-beasts.com>

---------

Co-authored-by: pdw-mb <pdw@mythic-beasts.com>

emontnemery committed:
Improve cloud migration (#92520)

* Improve cloud migration

* Tweak

* Use entity_ids func

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

DDanii committed:
Fix transmission error handling (#91548)

* transmission error handle fix

* added unexpected case tests

eavanvalkenburg committed:
Fix for SIA Code not being handled well (#92469)

* updated sia requirements

* updates because of changes in package

* linting and other small fixes

* fix for unknown code

* added same to alarm_control_panel

balloob committed:
Re-run expose entities migration if first time failed (#92564)

* Re-run expose entities migration if first time failed

* Count number of exposed entities

* Add tests

---------

Co-authored-by: Erik <erik@montnemery.com>

emontnemery committed:
Bump hatasmota to 0.6.5 (#92585)

* Bump hatasmota to 0.6.5

* Fix tests

epenet committed:
Fix hassio get_os_info retry (#92569)

balloob committed:
Bumped version to 2023.5.2

bdraco commented:
If we are waiting for https://github.com/home-assistant/core/pull/92617, https://github.com/home-assistant/core/pull/92551 is ready now as well

bdraco committed:
Improve reliability of ONVIF subscription renewals (#92551)

* Improve reliablity of onvif subscription renewals

upstream changelog: https://github.com/hunterjm/python-onvif-zeep-async/compare/v2.0.0...v2.1.0

* ```
Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/onvif/client.py", line 75, in _async_wrap_connection_error_retry
    return await func(*args, **kwargs)
  File "/Users/bdraco/home-assistant/homeassistant/components/onvif/event.py", line 441, in _async_call_pullpoint_subscription_renew
    await self._pullpoint_subscription.Renew(SUBSCRIPTION_RELATIVE_TIME)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/zeep/proxy.py", line 64, in __call__
    return await self._proxy._binding.send_async(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/zeep/wsdl/bindings/soap.py", line 156, in send_async
    response = await client.transport.post_xml(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/zeep/transports.py", line 235, in post_xml
    response = await self.post(address, message, headers)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/zeep/transports.py", line 220, in post
    response = await self.client.post(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_client.py", line 1845, in post
    return await self.request(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_client.py", line 1530, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_client.py", line 1617, in send
    response = await self._send_handling_auth(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_client.py", line 1645, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_client.py", line 1682, in _send_handling_redirects
    response = await self._send_single_request(request)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_client.py", line 1719, in _send_single_request
    response = await transport.handle_async_request(request)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_transports/default.py", line 352, in handle_async_request
    with map_httpcore_exceptions():
  File "/opt/homebrew/Cellar/python@3.10/3.10.10_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/contextlib.py", line 153, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadTimeout
```

* adjust timeouts for slower tplink cameras

* tweak

* more debug

* tweak

* adjust message

* tweak

* Revert "tweak"

This reverts commit 10ee2a8de70e93dc5be85b1992ec4d30c2188344.

* give time in seconds

* revert

* revert

* Update homeassistant/components/onvif/event.py

* Update homeassistant/components/onvif/event.py

joostlek committed:
Lower scan interval for OpenSky (#92593)

* Lower scan interval for opensky to avoid hitting rate limit

* Lower scan interval for opensky to avoid hitting rate limit

* Update homeassistant/components/opensky/sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/opensky/sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

bramkragten committed:
Update frontend to 20230503.3 (#92617)

bdraco committed:
Fix duplicate ONVIF sensors (#92629)

Some cameras do not configure the video source correctly
when using webhooks but work fine with PullPoint which
results in duplicate sensors

bdraco committed:
Fix missing ONVIF events when switching from PullPoint to webhooks (#92627)

We now let the PullPoint subscription expire instead of explicitly
unsubscribing when pausing the subscription. We will still unsubscribe
it if Home Assistant is shutdown or the integration is reloaded

Some cameras will cancel ALL subscriptions when we do an unsubscribe
so we want to let the PullPoint subscription expire instead
of explicitly cancelling it.

bdraco commented:
mysql test failure is unrelated. It looks like a race bug in freezegun
```
023-05-05T19:14:25.4121140Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 878, in _guarded_process_one_task_or_recover
2023-05-05T19:14:25.4121566Z     self._process_one_task_or_recover(task)
2023-05-05T19:14:25.4121975Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 889, in _process_one_task_or_recover
2023-05-05T19:14:25.4122360Z     return task.run(self)
2023-05-05T19:14:25.4122729Z   File "/home/runner/work/core/core/homeassistant/components/recorder/tasks.py", line 281, in run
2023-05-05T19:14:25.4123113Z     instance._process_one_event(self.event)
2023-05-05T19:14:25.4123529Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 1002, in _process_one_event
2023-05-05T19:14:25.4123932Z     self._commit_event_session_or_retry()
2023-05-05T19:14:25.4124341Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 1145, in _commit_event_session_or_retry
2023-05-05T19:14:25.4124832Z     self._commit_event_session()
2023-05-05T19:14:25.4125296Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 1165, in _commit_event_session
2023-05-05T19:14:25.4125661Z     session.commit()
2023-05-05T19:14:25.4126419Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1905, in commit
2023-05-05T19:14:25.4126811Z     trans.commit(_to_root=True)
2023-05-05T19:14:25.4127073Z   File "<string>", line 2, in commit
2023-05-05T19:14:25.4127613Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/state_changes.py", line 137, in _go
2023-05-05T19:14:25.4128005Z     ret_value = fn(self, *arg, **kw)
2023-05-05T19:14:25.4128751Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1220, in commit
2023-05-05T19:14:25.4129161Z     self._prepare_impl()
2023-05-05T19:14:25.4129446Z   File "<string>", line 2, in _prepare_impl
2023-05-05T19:14:25.4129986Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/state_changes.py", line 137, in _go
2023-05-05T19:14:25.4130540Z     ret_value = fn(self, *arg, **kw)
2023-05-05T19:14:25.4131093Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1195, in _prepare_impl
2023-05-05T19:14:25.4131483Z     self.session.flush()
2023-05-05T19:14:25.4132003Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 4153, in flush
2023-05-05T19:14:25.4132375Z     self._flush(objects)
2023-05-05T19:14:25.4132868Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 4289, in _flush
2023-05-05T19:14:25.4133262Z     with util.safe_reraise():
2023-05-05T19:14:25.4133806Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/util/langhelpers.py", line 147, in __exit__
2023-05-05T19:14:25.4134224Z     raise exc_value.with_traceback(exc_tb)
2023-05-05T19:14:25.4134772Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 4250, in _flush
2023-05-05T19:14:25.4135159Z     flush_context.execute()
2023-05-05T19:14:25.4135887Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/unitofwork.py", line 467, in execute
2023-05-05T19:14:25.4136441Z     rec.execute(self)
2023-05-05T19:14:25.4137157Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/unitofwork.py", line 644, in execute
2023-05-05T19:14:25.4137781Z     util.preloaded.orm_persistence.save_obj(
2023-05-05T19:14:25.4138361Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/persistence.py", line 60, in save_obj
2023-05-05T19:14:25.4138730Z     for (
2023-05-05T19:14:25.4139275Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/persistence.py", line 221, in _organize_states_for_save
2023-05-05T19:14:25.4139738Z     for state, dict_, mapper, connection in _connections_for_states(
2023-05-05T19:14:25.4140375Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/persistence.py", line 1726, in _connections_for_states
2023-05-05T19:14:25.4140883Z     connection = uowtransaction.transaction.connection(base_mapper)
2023-05-05T19:14:25.4141251Z   File "<string>", line 2, in connection
2023-05-05T19:14:25.4141791Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/state_changes.py", line 137, in _go
2023-05-05T19:14:25.4142184Z     ret_value = fn(self, *arg, **kw)
2023-05-05T19:14:25.4142719Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 965, in connection
2023-05-05T19:14:25.4143151Z     return self._connection_for_bind(bind, execution_options)
2023-05-05T19:14:25.4143497Z   File "<string>", line 2, in _connection_for_bind
2023-05-05T19:14:25.4144050Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/state_changes.py", line 137, in _go
2023-05-05T19:14:25.4144447Z     ret_value = fn(self, *arg, **kw)
2023-05-05T19:14:25.4145034Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1098, in _connection_for_bind
2023-05-05T19:14:25.4145494Z     conn = self._parent._connection_for_bind(bind, execution_options)
2023-05-05T19:14:25.4145831Z   File "<string>", line 2, in _connection_for_bind
2023-05-05T19:14:25.4146383Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/state_changes.py", line 137, in _go
2023-05-05T19:14:25.4146772Z     ret_value = fn(self, *arg, **kw)
2023-05-05T19:14:25.4147465Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1150, in _connection_for_bind
2023-05-05T19:14:25.4147884Z     transaction = conn.begin()
2023-05-05T19:14:25.4148413Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 849, in begin
2023-05-05T19:14:25.4148979Z     self._transaction = RootTransaction(self)
2023-05-05T19:14:25.4149534Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2667, in __init__
2023-05-05T19:14:25.4149906Z     self._connection_begin_impl()
2023-05-05T19:14:25.4150473Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2685, in _connection_begin_impl
2023-05-05T19:14:25.4150889Z     self.connection._begin_impl(self)
2023-05-05T19:14:25.4151436Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1087, in _begin_impl
2023-05-05T19:14:25.4151841Z     self._log_info("BEGIN (implicit)")
2023-05-05T19:14:25.4152382Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 193, in _log_info
2023-05-05T19:14:25.4152803Z     self.engine.logger.info(message, *arg, **kw)
2023-05-05T19:14:25.4153202Z   File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/logging/__init__.py", line 1477, in info
2023-05-05T19:14:25.4153566Z     self._log(INFO, msg, args, **kwargs)
2023-05-05T19:14:25.4153938Z   File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/logging/__init__.py", line 1622, in _log
2023-05-05T19:14:25.4154346Z     record = self.makeRecord(self.name, level, fn, lno, msg, args,
2023-05-05T19:14:25.4154770Z   File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/logging/__init__.py", line 1591, in makeRecord
2023-05-05T19:14:25.4155198Z     rv = _logRecordFactory(name, level, fn, lno, msg, args, exc_info, func,
2023-05-05T19:14:25.4155621Z   File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/logging/__init__.py", line 288, in __init__
2023-05-05T19:14:25.4155963Z     ct = time.time()
2023-05-05T19:14:25.4156465Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/freezegun/api.py", line 176, in fake_time
2023-05-05T19:14:25.4156849Z     current_time = get_current_time()
2023-05-05T19:14:25.4157589Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/freezegun/api.py", line 170, in get_current_time
2023-05-05T19:14:25.4158223Z     return freeze_factories[-1]()
2023-05-05T19:14:25.4158588Z IndexError: list index out of range
2023-05-05T19:14:25.4159843Z ERROR:homeassistant.components.recorder.core:SQLAlchemyError error processing task EventTask(event=<Event EVENT_TEST[L]: test_attr=5, test_attr_10=nice>): This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: list index out of range (Background on this error at: https://sqlalche.me/e/20/7s2a)
2023-05-05T19:14:25.4160651Z Traceback (most recent call last):
2023-05-05T19:14:25.4161068Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 889, in _process_one_task_or_recover
2023-05-05T19:14:25.4161447Z     return task.run(self)
2023-05-05T19:14:25.4161823Z   File "/home/runner/work/core/core/homeassistant/components/recorder/tasks.py", line 281, in run
2023-05-05T19:14:25.4162214Z     instance._process_one_event(self.event)
2023-05-05T19:14:25.4162631Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 1002, in _process_one_event
2023-05-05T19:14:25.4163009Z     self._commit_event_session_or_retry()
2023-05-05T19:14:25.4163431Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 1145, in _commit_event_session_or_retry
2023-05-05T19:14:25.4163821Z     self._commit_event_session()
2023-05-05T19:14:25.4164375Z   File "/home/runner/work/core/core/homeassistant/components/recorder/core.py", line 1165, in _commit_event_session
2023-05-05T19:14:25.4165183Z     session.commit()
2023-05-05T19:14:25.4171596Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1905, in commit
2023-05-05T19:14:25.4172248Z     trans.commit(_to_root=True)
2023-05-05T19:14:25.4172814Z   File "<string>", line 2, in commit
2023-05-05T19:14:25.4173615Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/state_changes.py", line 103, in _go
2023-05-05T19:14:25.4174430Z     self._raise_for_prerequisite_state(fn.__name__, current_state)
2023-05-05T19:14:25.4175105Z   File "/home/runner/work/core/core/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 905, in _raise_for_prerequisite_state
2023-05-05T19:14:25.4175576Z     raise sa_exc.PendingRollbackError(
2023-05-05T19:14:25.4176815Z sqlalchemy.exc.PendingRollbackError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: list index out of range (Background on this error at: https://sqlalche.me/e/20/7s2a)
```
